docs/guides/integration-dsp-no-sdk.md

```diff
@@ -0,0 +1,128 @@
+- Parse the response.
+
+You'll call the following API endpoint:
+
+```
+/v2/key/bidstream
+```
+
+Refresh the keys once at startup, and then periodically (recommended refresh interval is hourly).
+
+The UID2 SDK for C#&nbsp;/&nbsp;.NET uses a `Refresh` function.
+
+For an implementation example that shows encrypting the request and decrypting the response, see [Encryption and Decryption Code Examples](../getting-started/gs-encryption-decryption.md#encryption-and-decryption-code-examples).
+
+The API response is in JSON format, and includes `site_data`, the list of domains or app names that are allowed for the site.
+
+To see how all the fields are parsed, refer to the UID2 SDK for C#&nbsp;/&nbsp;.NET parse function: see [KeyParser.cs, lines 41-74](https://github.com/IABTechLab/uid2-client-net/blob/6ac53b106301e431a4aada3cbfbb93f8164ff7be/src/UID2.Client/KeyParser.cs#L41-L74).
+
+After decrypting the token into a raw UID2, if the token was generated on the client side, you can use the information in `site_data` to verify that a specific domain or app name is on the list of names allowed for it. For details, see [Verify the Domain or App Name](#verify-the-domain-or-app-name).
+
+## Decrypt UID2 Tokens into Raw UID2s
+
+When you have current keys, you'll be able to decrypt a UID2 token into a raw UID2. You also need to check several conditions to make sure that the token is eligible for use in the bidstream.
+
+You'll need to complete the following steps:
+
+1. [Check the token version](#check-the-token-version).
+
+1. [Decrypt the token](#decrypt-the-token).
+
+1. [Make sure the token has not expired](#make-sure-the-token-has-not-expired).
+
+1. [Verify that the domain or app name is valid](#verify-the-domain-or-app-name).
+
+The UID2 SDK for C# / .NET uses a `DecryptTokenIntoRawUid` function to perform these steps: see [BidstreamClient.cs, Line 15](https://github.com/IABTechLab/uid2-client-net/blob/6ac53b106301e431a4aada3cbfbb93f8164ff7be/src/UID2.Client/BidstreamClient.cs#L15).
+
+### Check the Token Version
+
+There are different UID2 token versions in current use, and later processing steps are a little different depending on whether the token version is v2 or a later version.
+
+You can check the token version using the first four characters of the token.
+
+To review detailed logic, see [UID2Encryption.cs, lines 36-50](https://github.com/IABTechLab/uid2-client-net/blob/6ac53b106301e431a4aada3cbfbb93f8164ff7be/src/UID2.Client/UID2Encryption.cs#L36-L50).
+
+### Decrypt the Token
+
+Use the master key and site key to decrypt the token. For a code example, refer to the `Decrypt` function: see [UID2Encryption.cs, line 29](https://github.com/IABTechLab/uid2-client-net/blob/6ac53b106301e431a4aada3cbfbb93f8164ff7be/src/UID2.Client/UID2Encryption.cs#L29). This function decrypts UID2 tokens into raw UID2s as part of the UID2 SDK for C# / .NET, and includes logic to handle different token versions.
+
+### Make Sure the Token Has Not Expired
```

Comment:
We're not just making sure it hasn't expired, we're also checking it has a valid lifetime (eg a token that has a lifetime of 31 days is invalid and rejected, even though it hasn't expired).

If the term `valid lifetime` is too confusing, we can say `valid timestamps`

Reply:
@jon8787 reworded, thx. This touches on something else... Slack thread.